qiskit/pulse/schedule.py

```diff
@@ -926,65 +478,1111 @@ def draw(self,
+                         'Apply ``qiskit.pulse.transforms.block_to_schedule`` function to this '
+                         'program to get ``Schedule`` representation supporting this method. '
+                         'This method is being deprecated.')
+
+    return wrapper
+
+
+class ScheduleBlock(PulseProgram):
+    """A quantum program *schedule block* with alignment policy that allows lazy scheduling
+    of instructions. The preferred usage is through the pulse builder.
+
+    This program representation doesn't track an explicit notion of timing, in contrast to the
+    ``Schedule`` representation. The timing of an instruction is managed by its relative position
+    and the ``transform`` argument. ``ScheduleBlock``s should be initialized with one of
+    the following transform policies:
+
+        - ``left`` ... Align instructions in the `as-soon-as-possible` manner.
+            Instructions are scheduled at the earliest possible time on the channel.
+        - ``right`` ... Align instructions in the `as-late-as-possible` manner.
+            Instructions are scheduled at the latest possible time on the channel.
+        - ``sequential`` ... Align instructions sequentially even though they are
+            allocated in different channels.
+        - ``equispaced`` ... Align instructions with equal interval within a specified duration.
+            Instructions on different channels are aligned sequentially.
+        - ``func`` ... Align instructions with arbitrary position within the given duration.
+            The position is specified by a callback function taking a pulse index ``j`` and
+            returning a fractional coordinate in [0, 1].
+
+    The overlap constraint of pulses, i.e. ``timeslots``, is not immediately evaluated,
+    and thus we can assign a parameter object to the instruction duration.
+    By using ``ScheduleBlock`` representation we can fully parametrize pulse waveform.
+    For example, Rabi schedule generator can be defined as
+
+    .. code-block:: python
+
+        duration = Parameter('rabi_dur')
+        amp = Parameter('rabi_amp')
+
+        block = ScheduleBlock()
+        rabi_pulse = pulse.Gaussian(duration=duration, amp=amp, sigma=duration/4)
+
+        block += Play(rabi_pulse, pulse.DriveChannel(0))
+        block += Call(measure_schedule)
+
+    Note that such waveform cannot be appended to the ``Schedule`` representation.
+
+    In the block representation, interval between two instructions can be
+    managed with ``Delay`` instruction. Because the schedule block lacks instruction
+    start time ``t0``, we cannot ``insert`` or ``shift`` target instruction.
+    In addition, stored instructions are not interchangable because the schedule block is
+    sensitive to the relative position of instructions.
+    Apart from these differences, the block representation can provide compatible
+    functionality with ``Schedule`` representation.
+    """
```

Comment:
```suggestion
    """A quantum program's *schedule block* with alignment policies that allow lazy the scheduling
    of instructions. This is the representation constructed by the pulse builder by default.

    This program representation doesn't track an explicit notion of timing, in contrast to the
    ``Schedule`` representation. The timing of an instruction is managed by its relative position
    and the stored ``transform`` argument. ``ScheduleBlock``s should be initialized with one of
    the following transform policies:

        - ``left`` ... Align instructions in the `as-soon-as-possible` manner.
            Instructions are scheduled at the earliest possible time on the channel.
        - ``right`` ... Align instructions in the `as-late-as-possible` manner.
            Instructions are scheduled at the latest possible time on the channel.
        - ``sequential`` ... Align instructions sequentially even though they are
            allocated in different channels.
        - ``equispaced`` ... Align instructions with equal interval within a specified duration.
            Instructions on different channels are aligned sequentially.
        - ``func`` ... Align instructions with arbitrary position within the given duration.
            The position is specified by a callback function taking a pulse index ``j`` and
            returning a fractional coordinate in [0, 1].

    The overlap constraint of pulses, i.e. ``timeslots``, is not immediately evaluated,
    and thus we can assign a parameter object to the instruction duration.
    By using ``ScheduleBlock`` representation we can fully parametrize pulse waveform.
    For example, Rabi schedule generator can be defined as

    .. code-block:: python

        duration = Parameter('rabi_dur')
        amp = Parameter('rabi_amp')

        block = ScheduleBlock()
        rabi_pulse = pulse.Gaussian(duration=duration, amp=amp, sigma=duration/4)

        block += Play(rabi_pulse, pulse.DriveChannel(0))
        block += Call(measure_schedule)

    Note that such waveform cannot be appended to the ``Schedule`` representation.

    In the block representation, the interval between two instructions can be
    managed with the ``Delay`` instruction. Because the schedule block lacks an instruction
    start time ``t0``, we cannot ``insert`` or ``shift`` the target instruction.
    In addition, stored instructions are not interchangable because the schedule block is
    sensitive to the relative position of instructions.
    Apart from these differences, the block representation can provide compatible
    functionality with ``Schedule`` representation.
    """
```